libs/src/sdk/api/developer-apps/DeveloperAppsAPI.ts

```diff
@@ -0,0 +1,111 @@
+
+import * as secp from '@noble/secp256k1'
+import type { AuthService, EntityManagerService } from '../../services'
+
+import { Action, EntityType } from '../../services/EntityManager/types'
+
+import { decodeHashId } from '../../utils/hashId'
+import { objectMissingValues } from '../../utils/object'
+import {
+  CREATE_DEVELOPER_APP_REQUIRED_VALUES,
+  DELETE_DEVELOPER_APP_REQUIRED_VALUES
+} from './constants'
+import type {
+  CreateDeveloperAppRequest,
+  DeleteDeveloperAppRequest
+} from './types'
+
+// Note (nkang): Eventually this will extend the generated DeveloperAppsAPI
+export class DeveloperAppsAPI {
+  constructor(
+    _config: Configuration,
+    private readonly entityManager: EntityManagerService,
+    private readonly auth: AuthService
+  ) {}
+
+  /**
+   * Create a developer app
+   */
+  async createDeveloperApp(requestParameters: CreateDeveloperAppRequest) {
+    const { name, userId, isPersonalAccess } = requestParameters
+    const metadataMissingValues = objectMissingValues(
+      requestParameters,
+      CREATE_DEVELOPER_APP_REQUIRED_VALUES
+    )
+    if (metadataMissingValues?.length) {
+      throw new Error(
+        `Create Developer App error - Metadata object is missing values: ${metadataMissingValues}`
+      )
+    }
+
+    const decodedUserId = decodeHashId(userId)
+    if (decodedUserId == null) {
+      throw new Error(`Create Developer App error - user id is invalid`)
+    }
+    const apiSecretRaw = secp.utils.randomPrivateKey()
+    const apiKeyRaw = secp.getPublicKey(apiSecretRaw, true)
+    const apiSecret = Buffer.from(apiSecretRaw).toString('hex')
+    const apiKey = Buffer.from(apiKeyRaw).toString('hex')
+    const response = await this.entityManager.manageEntity({
+      userId: decodedUserId,
+      entityType: EntityType.DELEGATE,
+      entityId: 0, // Contract requires uint, but we don't actually need this field for this action. Just use 0.
+      action: Action.CREATE,
+      metadata: JSON.stringify({
+        name,
+        address: apiKey,
+        is_personal_access: isPersonalAccess ?? false
+      }),
+      auth: this.auth
+    })
+    // TODO(nkang): If is_personal_access is true, create the grant from user to developer app. Waiting until SDK confirmer is implemented, so for now we can trigger the grant creation in client.
+    const txReceipt = response.txReceipt
+    return {
+      blockHash: txReceipt.blockHash,
+      blockNumber: txReceipt.blockNumber,
+      apiKey,
+      apiSecret,
+      error: false
+    }
+  }
+
+  /**
+   * Delete a developer app
+   */
+  async deleteDeveloperApp(requestParameters: DeleteDeveloperAppRequest) {
+    const { userId, appApiKey } = requestParameters
+    const metadataMissingValues = objectMissingValues(
+      requestParameters,
+      DELETE_DEVELOPER_APP_REQUIRED_VALUES
+    )
+    if (metadataMissingValues?.length) {
+      throw new Error(
+        `Delete Developer App error - Metadata object is missing values: ${metadataMissingValues}`
```

Comment:
might be worth normalizing how we handle errors - see how I handle them in ChatsApi and maybe we could break out a helper util for asserts like this

Reply:
will do

Reply:
let's revisit this after the zod stuff